sdk/textanalytics/Azure.AI.TextAnalytics/src/Validation.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using static Azure.AI.TextAnalytics.TextAnalyticsClientOptions;
+
+namespace Azure.AI.TextAnalytics
+{
+    internal static class Validation
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void SupportsOperation(string name, ServiceVersion added, ServiceVersion current)
```

Comment:
Should we include the client name as well?  It might make it quicker to diagnose the failure.

Reply:
That'd be more to pass down from the caller which, on its own isn't a problem, but as asked above: does it add value? The exception would clearly show what was called where. And we only have one public client anyway.

Reply:
I'd like a general pattern and many libraries will have multiple clients - so I think it adds value.

Reply:
I did this and isolated all the callees to a "Validation" folder, though the namespace doesn't reflect that so no addiitonal `using`s are needed and might have to be removed later.